experimental/segmenter/src/complex.rs

```diff
@@ -13,9 +13,26 @@ use crate::DictionarySegmenter;
 #[cfg(feature = "lstm")]
 use crate::lstm::{get_best_lstm_model, LstmSegmenter};
 
+fn get_best_dictionary_paylod(
+    input: u32,
+    payload: [Option<&DataPayload<UCharDictionaryBreakDataV1Marker>>; 5],
```

Comment:
Suggestion: `[Option<&DataPayload<UCharDictionaryBreakDataV1Marker>>; 5]` isn't a nice type, and it's very easy to populate the fields in an incorrect order. Use a struct instead:

```rust
struct Dictionaries {
  khmer: Option<&DataPayload<UCharDictionaryBreakDataV1Marker>>,
  lao: Option<&DataPayload<UCharDictionaryBreakDataV1Marker>>,
  burmese: Option<&DataPayload<UCharDictionaryBreakDataV1Marker>>,
  thai: Option<&DataPayload<UCharDictionaryBreakDataV1Marker>>,
  cj: Option<&DataPayload<UCharDictionaryBreakDataV1Marker>>,
}

impl Dictionaries {
  fn best(&self, input: u32) -> Option<&DataPayload<UCharDictionaryBreakDataV1Marker>> {
    match get_language(input) {
        Language::Khmer => self.khmer,
        Language::Lao => self.lao,
        Language::Burmese => self.burmese,
        Language::Thai => self.thai,
        Language::ChineseOrJapanese => self.cj,
        _ => None,
    }

  }
}